### DISCUSSION TITLE
NotSelected validator for Select?

### DISCUSSION THREAD

**Discussion by sturlath** [Megabit/Blazorise#2827] [Q&A]
There are some [inbuilt validation rules](https://blazorise.com/docs/components/validation/#validation-rules) but there is no **IsNotSelected** (or IsSelected) rule for Select? 

Is there a room for that?

Now I guess I need a validation event like this to kick off validation if a required dropdown has not been selected from.

```csharp
private void ValidateSelected(ValidatorEventArgs e)
{
   if(MyModel.MyBoundProperty != null) //or != Date.Min,0, etc
   {
       e.Status =  ValidationStatus.Success;
    }
    else
    {
        e.Status =  ValidationStatus.Error;
    }
}
```

**Comment by David-Moreira** [comment]
Hello,
when using the Validation Delegate approach, you should access the value being validated through the `e.Value` like this and then applying the appropriate validation logic:

```
  void ValidateEmail( ValidatorEventArgs e )
  {
      var email = Convert.ToString( e.Value );

      e.Status = string.IsNullOrEmpty( email ) ? ValidationStatus.None :
          email.Contains( "@" ) ? ValidationStatus.Success : ValidationStatus.Error;
  }
```

For standard Selected / IsNotSelected validation this one might be usefull? Depends on how you actually validate if something is selected.
`<Validation Validator="@ValidationRule.IsNotEmpty">` 

or you can also make your own general Validators that apply to your use cases like for example a generic one based on a boolean value:
```
  void ValidateCheck( ValidatorEventArgs e )
  {
      e.Status = bool.TryParse( e.Value.ToString(), out bool result ) && result
                  ? ValidationStatus.Success
                  : ValidationStatus.Error;

  }
```

There's also a few examples in the docs:
https://blazorise.com/docs/components/validation/#example

**Reply by stsrki** [reply]
I agree that we should have a built-in validator for Select components with overloads for standard data types like int, long, string, etc. The only downside is that we cannot know for sure what is the invalid value since for the end-user it can be anything. For example, it can be `0`, `-1`, `string.empty`, `"empty"`, `null`, etc. We can only assume some of them to be invalid values.
